### PR TITLE
Throw ANE for null TextEncoderSettings to TextEncoderSettings ctor

### LINKED ISSUE #23600: System.Text.Encodings.Web.TextEncoder throws NRE on constructing copy
```C#
TextEncoder copyFrom = null;
TextEncoder copy = new TextEncoder(copyFrom); // Throws NullReferenceException
```

Should really throw `ArgumentNullException`.

### PR DESCRIPTION
**Pull request by JonHanna**
Fixes #23600

Also further test to bring coverage for that class up to 100%.

### CODE REVIEW

**Comment by JonHanna** [comment]
Test Windows x64 Debug Build

File: src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoderSettings.cs [hunk 1]
```diff
@@ -28,6 +28,11 @@ public TextEncoderSettings()
         /// </summary>
         public TextEncoderSettings(TextEncoderSettings other)
         {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
```
Comment (karelz) [inline src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoderSettings.cs]: Does it throw same exception on Desktop?

Comment (tarekgh) [reply src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoderSettings.cs]: @karelz  Desktop doesn't have this library.

**Comment by tarekgh** [comment]
LGTM. 

@karelz if you don't have any other questions or comments, please go ahead and merge this one. Thanks.

**Comment by karelz** [comment]
OK, technically breaking against Core 1.0/1.1, but I think we can live with that.

**Comment by karelz** [comment]
Merged, thanks @JonHanna!

**Comment by JonHanna** [comment]
@karelz `NullReferenceException` is specifically mentioned in https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#behavioral-changes as something that can be replaced with another exception. Presumably since one generally doesn't depend on catching it, but finds why it happened and fix that. `ArgumentNullException` makes that task of finding and fixing a little bit easier.

**Pull request merged**